api/v1alpha1/envoygateway_types.go

```diff
@@ -202,6 +202,13 @@ type EnvoyGatewayKubernetesProvider struct {
 	// +optional
 	RateLimitDeployment *KubernetesDeploymentSpec `json:"rateLimitDeployment,omitempty"`
 
+	// RateLimitDaemonset defines the desired state of the Envoy ratelimit daemonset resource.
+	// If unspecified, default settings for the managed Envoy ratelimit daemonset resource
+	// are applied.
+	//
+	// +optional
+	RateLimitDaemonset *KubernetesDaemonSetSpec `json:"rateLimitDaemonset,omitempty"`
```

Comment:
I might be mistaken but they should be able to exist in either mode alongside each other. That said, I can't think of a use-case for running mixed (maybe EG daemonset and ratelimit deployment?) so adding a check to prevent that makes sense. I'll look into that more tomorrow if you'd like.

Reply:
> the EG and RL pod will be running on each node, does it mean the global rate limiting becomes per-node rate limiting?

I'm not currently using the feature but reading through the docs and code leads me to think that nothing would change. RL pods all share the same Redis backend so it shouldn't become a per-node rate limit or adjust any logic.

Ref:
- https://gateway.envoyproxy.io/latest/tasks/traffic/local-rate-limit/
- https://gateway.envoyproxy.io/latest/tasks/traffic/global-rate-limit/